lib/base/configobject.cpp

```diff
@@ -565,7 +565,7 @@ void ConfigObject::RestoreObjects(const String& filename, int attributeTypes)
 		if (srs != StatusNewItem)
 			continue;
 
-		upq.Enqueue(std::bind(&ConfigObject::RestoreObject, message, attributeTypes));
+		upq.Enqueue([&message, &attributeTypes](){ RestoreObjects(message, attributeTypes); });
```

Comment:
Explain with your own words what’s the problem with capturing `message` by reference.

Reply:
If a lambda function tries to access or modify a local variable captured by reference that is no longer in scope, i.e. that has been destroyed due to stack unwinding, a crash may occur. And if I'm not mistaken this lambda function does exactly that, because the ``message`` variable will certainly be modified and this then lose it's scope and would likely cause a ``segfault`` and while this procedure is repeated over and over again in an infinite for loop without recreating the variable, an unexpected ``Icinga2 crash💥`` is inevitable.

Reply:
Your first sentence already names it. Have a look for more of such in your PR.